component/autoscaler.jsonnet

```diff
@@ -1,11 +1,61 @@
 local com = import 'lib/commodore.libjsonnet';
+local espejote = import 'lib/espejote.libsonnet';
 local kap = import 'lib/kapitan.libjsonnet';
 local kube = import 'lib/kube.libjsonnet';
 local inv = kap.inventory();
 
 local params = inv.parameters.openshift4_nodes;
 local metrics = import 'autoscaling-metrics.libsonnet';
 
+local autoscalerArgsPatch = if params.autoscaling.addAutoscalerArgs != null && std.length(params.autoscaling.addAutoscalerArgs) > 0 then
+  assert std.member(inv.applications, 'espejote') : 'The addAutoscalerArgs patch depends on espejote';
+  {
+    autoscaler_inject_args_admission: espejote.admission('autoscaler-inject-args', params.machineApiNamespace) {
+      metadata+: {
+        annotations+: {
+          description: "Injects autoscaler arguments to the default autoscaler pod in the openshift-machine-api namespace.\nArguments are taken from flagsToAdd.\n\nThe patch blindly adds the arguments without trying to replace them if they already exist.\nI debated replacing them but we won't be able to guess all upstream changes in the args array and our args parsing might fail anyways.\nI prefer a simpler patch that fails fast to a convoluted args array merge.\n",
+        },
+      },
+      spec: {
+        mutating: true,
+        webhookConfiguration: {
+          rules: [
+            {
+              apiGroups: [
+                '',
+              ],
+              apiVersions: [
+                '*',
+              ],
+              operations: [
+                'CREATE',
+              ],
+              resources: [
+                'pods',
+              ],
+            },
+          ],
+          objectSelector: {
```

Comment:
Do I read this right that we would intercept pods in any namespace if they have these labels?

Reply:
Admission are namespaced and only intercept in their respective namespaces.